server/lib/mutex.ts

```diff
@@ -0,0 +1,41 @@
+import debugLib from 'debug';
+
+import logger from './logger';
+import { createRedisClient } from './redis';
+import { sleep } from './utils';
+
+const debug = debugLib('mutex');
+
+/**
+ * Cache based, service-wide mutex implementation.
+ */
+export async function lockUntilResolved<T>(
+  key: string,
+  until: () => Promise<T>,
+  { lockAcquireTimeoutMs = 15 * 1000, unlockTimeoutMs = 60 * 1000 } = {},
+): Promise<T> {
+  const redis = await createRedisClient();
```

Comment:
Have you considered targeting another specific redis instance?

Reply:
We've seen keys being flushed very quickly.

It **may** work.

Reply:
You're correct. I think the nature of the other cache is very different from this one. Those are discardable and these are not.